iotdb-core/ainode/iotdb/ainode/core/inference/inference_request_pool.py

```diff
@@ -107,42 +111,77 @@ def _requests_activate_loop(self):
-        # TODO: We need a batcher to accelerate the concurrent inference
-        for request in requests:
+        all_requests: list[InferenceRequest] = self._request_scheduler.schedule_step()
+
+        grouped_requests = defaultdict(list)
+        for req in all_requests:
+            key = (req.inputs.shape[1], req.max_new_tokens)
+            grouped_requests[key].append(req)
+        grouped_requests = list(grouped_requests.values())
+
+        for requests in grouped_requests:
+            batch_inputs = self._batcher.batch_request(requests).to(self.device)
             if self.model_id == "sundial":
-                request.inputs = request.inputs.to(self.device)
-                output = self._model.generate(
-                    request.inputs,
-                    max_new_tokens=request.max_new_tokens,
+                batch_output = self.model.generate(
+                    batch_inputs,
+                    max_new_tokens=requests[0].max_new_tokens,
                     num_samples=10,
                     revin=True,
                 )
-                request.output_tensor = request.output_tensor.to(self.device)
-                request.write_step_output(output[0].mean(dim=0))
+
+                offset = 0
+                for request in requests:
+                    request.output_tensor = request.output_tensor.to(self.device)
+                    b = request.batch_size
+                    output_i = batch_output[offset : offset + b]
+                    offset += b
+                    # ! Here we only considered the case where batchsize=1 in one request. If multi-variable adaptation is required in the future, modifications may be needed here
+                    request.write_step_output(output_i[0].mean(dim=0))
+
+                    request.inference_pipeline.post_decode()
+                    if request.is_finished():
+                        request.inference_pipeline.post_inference()
+                        self._logger.debug(
+                            f"[Inference][Device-{self.device}][Pool-{self.pool_id}][ID-{request.req_id}] Request is finished"
+                        )
+                        # ensure the output tensor is on CPU before sending to result queue
+                        request.output_tensor = request.output_tensor.cpu()
+                        self._finished_queue.put(request)
+                    else:
+                        self._logger.debug(
+                            f"[Inference][Device-{self.device}][Pool-{self.pool_id}][ID-{request.req_id}] Request is not finished, re-queueing"
+                        )
+                        self._waiting_queue.put(request)
+
             elif self.model_id == "timer_xl":
-                request.inputs = request.inputs.to(self.device)
-                output = self._model.generate(
-                    request.inputs,
-                    max_new_tokens=request.max_new_tokens,
+                batch_output = self.model.generate(
```

Comment:
Plz append concurrent inference test for `timer_xl` in `AINodeConcurrentInferenceIT`.